src/main/java/LadderGame/Ladder.java

```diff
@@ -1,58 +1,90 @@
 package LadderGame;
 
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
 import java.util.Scanner;
+import java.util.stream.Collectors;
 
 public class Ladder {
-    private int numPeople;
-    private int height;
-    private String[][] ladder;
+    private int ladder_height;
+    private List<List<String>> ladder_set;
+    private List<String> names;
 
     // 초기값 세팅
     public Ladder() {
         input();
-        ladder = new String[height][numPeople * 2 - 1];
+        ladder_set = new ArrayList<>();
     }
 
     // 입력 부분
-    public void input(){
+    public void input() {
         Scanner in = new Scanner(System.in);
-        System.out.print("참여할 사람은 몇 명인가요?\n");
-        this.numPeople = in.nextInt();
+
+        System.out.print("참여할 사람 이름을 입력하세요. (이름은 쉼표(,)로 구분하세요)\n");
+        this.names = parsingName(in.nextLine());
+
         System.out.print("최대 사다리 높이는 몇 개인가요?\n");
-        this.height = in.nextInt();
+        this.ladder_height = in.nextInt();
+    }
+
+    // 이름 분리
+    public List<String> parsingName(String str) {
+        // map을 통하여 5글자 제한 기능 구현
+        return Arrays.stream(str.split(","))
+                .map(s -> s.length() > 5 ? s.substring(0, 5) : s)
+                .collect(Collectors.toList());
     }
 
     // 사다리 설정
-    public void set(int line) {
-        for (int j = 0; j < numPeople * 2 - 1; j++) {
-            ladder[line][j] = LineCheck(j);
+    public void setLadder(int line) {
+        List<String> temp_List = new ArrayList<>();
+
+        for (int j = 0; j < names.size() * 2 - 1; j++) {
+            String before_line = j < 2 ? "     " : temp_List.get(j - 2);
+            temp_List.add(lineCheck(j, before_line));
+        }
+        ladder_set.add(temp_List);
+
+        if (line + 1 < ladder_height) {
+            setLadder(line + 1);
         }
+    }
 
-        if (line + 1 < height) {
-            set(line + 1);
+    // 이름 출력
+    public void namePrint() {
+        for (String temp : names) {
+            System.out.print(temp);
+            spacePrint(temp.length());
+        }
+        System.out.println();
+    }
+
+    // 빈 공간 출력
+    public void spacePrint(int length) {
+        for (int i = 6 - length; i > 0; i--) {
+            System.out.print(" ");
         }
     }
 
     // 사다리 내부 값
-    public String LineCheck(int check) {
+    public String lineCheck(int check, String before_line) {
```

Comment:
```suggestion
    public String checkLine(int check, String before_line) {
```

메소드는 동사로 시작하는 게 좋습니다.
객체.메소드(매개변수)
주어.동사(목적어 | 보어) 느낌으로.